src/engine.rs

```diff
@@ -705,110 +713,110 @@ impl Reedline {
         self.prompt_widget.origin = origin;
     }
 
-    fn handle_event(&mut self, prompt: &dyn Prompt, event: Event) -> io::Result<Option<Signal>> {
+    fn handle_event(
+        &mut self,
+        prompt: &dyn Prompt,
+        event: ReedlineEvent,
+    ) -> io::Result<Option<Signal>> {
         match event {
-            Event::Key(KeyEvent { code, modifiers }) => {
-                match (modifiers, code) {
-                    (KeyModifiers::NONE, KeyCode::Tab) => {
-                        let mut line_buffer = self.editor.line_buffer();
-                        self.tab_handler.handle(&mut line_buffer);
-                        Ok(None)
-                    }
-                    (KeyModifiers::CONTROL, KeyCode::Char('d')) => {
-                        if self.editor.is_empty() {
-                            self.editor.reset_olds();
-                            Ok(Some(Signal::CtrlD))
-                        } else if let Some(binding) = self.find_keybinding(modifiers, code) {
-                            self.run_edit_commands(&binding);
-                            Ok(None)
-                        } else {
-                            Ok(None)
-                        }
-                    }
-                    (KeyModifiers::CONTROL, KeyCode::Char('c')) => {
-                        if let Some(binding) = self.find_keybinding(modifiers, code) {
-                            self.run_edit_commands(&binding);
-                        }
-                        self.editor.reset_olds();
-                        Ok(Some(Signal::CtrlC))
-                    }
-                    (KeyModifiers::CONTROL, KeyCode::Char('l')) => {
-                        self.editor.reset_olds();
-                        Ok(Some(Signal::CtrlL))
+            ReedlineEvent::HandleTab => {
+                let mut line_buffer = self.editor.line_buffer();
+                self.tab_handler.handle(&mut line_buffer);
+                Ok(None)
+            }
+            ReedlineEvent::CtrlD => {
+                if self.editor.is_empty() {
+                    self.editor.reset_olds();
+                    Ok(Some(Signal::CtrlD))
+                // } else if let Some(binding) = self.find_keybinding(modifiers, code) {
+                //     self.run_edit_commands(&binding);
+                //     Ok(None)
```

Comment:
The rationale to remove this was that input_parsing will handle conversion into what operation we want to perform and we care only about the end result here (i.e. what to do with the given action). 
Handling if keybindings are different is the responsibility of the input_parsing stage.